securedpreferencestore/src/main/java/devliving/online/securedpreferencestore/SecuredPreferenceStore.java

```diff
@@ -52,20 +53,49 @@ public static void setRecoveryHandler(RecoveryHandler recoveryHandler) {
         SecuredPreferenceStore.mRecoveryHandler = recoveryHandler;
     }
 
-    synchronized public static SecuredPreferenceStore getSharedInstance(Context appContext) throws IOException, CertificateException, NoSuchAlgorithmException, InvalidKeyException, UnrecoverableEntryException, InvalidAlgorithmParameterException, NoSuchPaddingException, NoSuchProviderException, KeyStoreException {
-        if (mInstance == null) {
-            mInstance = new SecuredPreferenceStore(appContext);
+    synchronized public static SecuredPreferenceStore getSharedInstance() {
+        if ( mInstance == null ) {
+            throw new IllegalStateException("Must call init() before using the store");
         }
 
         return mInstance;
     }
 
+    /**
+     * Must call this before using the store, otherwise the encryption manager will not be setup.
+     * This method allows getSharedInstance() to be called without having to handle the exceptions each time.
+     * Call this when first setting up the store in Application or your launching Activity and handle errors accordingly.
+     *
+     * @throws IOException
+     * @throws CertificateException
+     * @throws NoSuchAlgorithmException
+     * @throws KeyStoreException
+     * @throws UnrecoverableEntryException
+     * @throws InvalidAlgorithmParameterException
+     * @throws NoSuchPaddingException
+     * @throws InvalidKeyException
+     * @throws NoSuchProviderException
+     */
+    public static void init( Context appContext) throws IOException, CertificateException, NoSuchAlgorithmException, KeyStoreException, UnrecoverableEntryException, InvalidAlgorithmParameterException, NoSuchPaddingException, InvalidKeyException, NoSuchProviderException {
+        if ( mInstance == null ) {
+            mInstance = new SecuredPreferenceStore(appContext);
+        }
+    }
+
+    private void checkInitCalled() {
```

Comment:
This method is unnecessary as there is no way to have a non-null `SecuredPreferenceStore` instance unless the `init` was called and successful.